frontend/packages/dev-console/src/components/import/ImportForm.tsx

```diff
@@ -101,18 +105,36 @@ const ImportForm: React.FC<ImportFormProps & StateProps> = ({
     const imageStream = builderImages[values.image.selected].obj;
 
     const {
+      application: { name: applicationName },
       project: { name: projectName },
     } = values;
 
+    const kNativeServiceUtil = (dockerImageRepository: string) => {
+      createKnativeService(applicationName, namespace, dockerImageRepository)
+        .then(() => {
+          actions.setSubmitting(false);
+          handleRedirect(projectName);
+        })
+        .catch((err) => {
+          actions.setSubmitting(false);
+          actions.setStatus({ submitError: err.message });
+        });
+    };
+
     const dryRunRequests: Promise<K8sResourceKind[]> = createResources(values, imageStream, true);
     dryRunRequests
       .then(() => {
         const requests: Promise<K8sResourceKind[]> = createResources(values, imageStream);
         return requests;
       })
-      .then(() => {
-        actions.setSubmitting(false);
-        handleRedirect(projectName);
+      .then((responses: K8sResourceKind[]) => {
+        // if serverless option is selected, create knative service.
+        if (values.serverless.trigger) {
```

Comment:
Move this if condition to `createResources` and minimize the changes to this file.

Reply:
done